iverilog 13.0 errors

Running `make` with the latest version of iverilog bundled with https://github.com/YosysHQ/oss-cad-suite-build/releases/tag/2025-02-20 gives the following error on Ubuntu 24.04.

```
riscv64-unknown-elf-gcc -march=rv32i_zicsr -mabi=ilp32 -Os -Wall -Wextra -Wl,-Bstatic,-T,sections.lds,--strip-debug -ffreestanding -nostdlib -o firmware.elf firmware.s vectors.s firmware.c
/usr/lib/gcc/riscv64-unknown-elf/13.2.0/../../../riscv64-unknown-elf/bin/ld: warning: firmware.elf has a LOAD segment with RWX permissions
riscv64-unknown-elf-objcopy -O verilog firmware.elf firmware.hex
/home/philip/Downloads/oss-cad-suite-linux-x64-20250220/oss-cad-suite/bin/iverilog -g2012 -o testbench -D STALL -D NERV_DBGREGS testbench.sv nerv.sv
nerv.sv:565: error: Unable to bind wire/reg/memory `irq_num' in `testbench.dut'
nerv.sv:625:      : A symbol with that name was declared here. Check for declaration after use.
nerv.sv:565: error: Unable to elaborate r-value: ((((running)&&(imem_valid))&&(!(irq_num)))&&((insn_opcode)==(OPCODE_SYSTEM)))?(insn_funct3['sd1:'sd0]):(2'd0)
nerv.sv:612: error: Net csr_hpm_event_sel['sd2:'sd0] is not defined in this context.
nerv.sv:1171:      : A symbol with that name was declared here. Check for declaration after use.
nerv.sv:613: error: Net csr_hpm_counter_sel['sd1] is not defined in this context.
nerv.sv:766:      : A symbol with that name was declared here. Check for declaration after use.
nerv.sv:614: error: Net csr_hpm_counterh_sel['sd1] is not defined in this context.
nerv.sv:967:      : A symbol with that name was declared here. Check for declaration after use.
nerv.sv:1052: error: Unable to bind wire/reg/memory `mem_rdata' in `testbench.dut'
nerv.sv:1087:      : A symbol with that name was declared here. Check for declaration after use.
6 error(s) during elaboration.
make: *** [Makefile:32: testbench] Error 6
```

```
/oss-cad-suite-linux-x64-20250220/oss-cad-suite/bin/iverilog -v
Icarus Verilog version 13.0 (devel) (s20250103-25-g99580cd05)
```


Reverting to iverilog 12.0 makes it work.

--- a/nerv.sv
+++ b/nerv.sv
@@ -399,6 +399,8 @@ module nerv #(
 	reg [4:0] mem_rd_reg;
 	reg [4:0] mem_rd_func;
 
+	reg [31:0] mem_rdata;
+
 	reg mem_rd_enable_q;
 	reg [4:0] mem_rd_reg_q;
 	reg [4:0] mem_rd_func_q;
@@ -528,6 +530,7 @@ module nerv #(
 	localparam MCAUSE_ECALL_M_MODE             = 32'h0000000b;
 
 	localparam IRQ_MASK = 32'hFFFF0888;
+	reg [4:0] irq_num;
 
 	// next write, next destination (rd) value & register
 	reg next_wr;
@@ -608,21 +611,21 @@ module nerv #(
 	wire [31:0] csr_``NAME``_next  = csr_``ARRAY``_next[(INDEX)*32 +: 32];  \
 	assign csr_``ARRAY``_sel[INDEX] = csr_``NAME``_sel;
 
-	// dummy out missing select lines
-	assign csr_hpm_event_sel[2:0] = 0;
-	assign csr_hpm_counter_sel[1] = 0;
-	assign csr_hpm_counterh_sel[1] = 0;
-
 `NERV_CSRS
 `undef NERV_CSR_REG_MRW
 `undef NERV_CSR_VAL_MRW
 `undef NERV_CSR_VAL_MRO
 `undef NERV_CSR_ARR_DEF
 `undef NERV_CSR_ARR_MRW
+
+	// dummy out missing select lines
+	assign csr_hpm_event_sel[2:0] = 0;
+	assign csr_hpm_counter_sel[1] = 0;
+	assign csr_hpm_counterh_sel[1] = 0;
+
 `endif // NERV_CSR
 
 	wire [31:0] irq_en;
-	reg [4:0] irq_num;
 	assign irq_en = irq & csr_mie_value;
 
 	// resolve interrupt priority
@@ -1139,7 +1142,6 @@ module nerv #(
 		end
 	end
 
-	reg [31:0] mem_rdata;
 `ifdef NERV_RVFI
 	reg next_rvfi_intr;
 	reg rvfi_trap_q;

